overlord/state/change.go

```diff
@@ -423,12 +429,32 @@ func (c *Change) Status() Status {
 	panic(fmt.Sprintf("internal error: cannot process change status: %v", statusStats))
 }
 
+// addNotice records an occurrence of a change-update notice for this change.
+// The notice key is set to the change ID.
+func (c *Change) addNotice() error {
+	opts := &AddNoticeOptions{
+		Data: map[string]string{"kind": c.Kind()},
+	}
+	_, err := c.state.AddNotice(nil, ChangeUpdateNotice, c.id, opts)
+	return err
+}
+
+func shouldSkipChangeUpdateNotice(old, new Status) bool {
+	// Skip alternating Doing->Do->Doing and Undoing->Undo->Undoing notices
+	return (old == new) || (old == DoingStatus && new == DoStatus) || (old == UndoingStatus && new == UndoStatus)
+}
+
 func (c *Change) notifyStatusChange(new Status) {
-	if c.lastObservedStatus == new {
-		return
+	if c.lastObservedStatus != new {
+		c.state.notifyChangeStatusChangedHandlers(c, c.lastObservedStatus, new)
+		c.lastObservedStatus = new
+	}
+	if !shouldSkipChangeUpdateNotice(c.lastRecordedNoticeStatus, new) {
+		if err := c.addNotice(); err != nil {
```

Comment:
I would add a comment before this:

// implies State.writing()

Reply:
good point, thanks! updated